pkg/cassandra/datacenter.go

```diff
@@ -209,21 +212,40 @@ func NewDatacenter(klusterKey types.NamespacedName, template *DatacenterConfig)
 
 	dc.Spec.Tolerations = template.Tolerations
 
+	setMcacDisabled(dc, template)
+
 	return dc, nil
 }
 
 // setMgmtAPIHeap sets the management API heap size on a CassandraDatacenter
 func setMgmtAPIHeap(dc *cassdcapi.CassandraDatacenter, heapSize *resource.Quantity) {
-	if dc.Spec.PodTemplateSpec == nil {
-		dc.Spec.PodTemplateSpec = &corev1.PodTemplateSpec{}
-	}
-
 	UpdateCassandraContainer(dc.Spec.PodTemplateSpec, func(c *corev1.Container) {
 		heapSizeInBytes := heapSize.Value()
 		c.Env = append(c.Env, corev1.EnvVar{Name: mgmtApiHeapSizeEnvVar, Value: fmt.Sprintf("%v", heapSizeInBytes)})
 	})
 }
 
+func isMcacEnabled(template *DatacenterConfig) bool {
+	if template.McacEnabled != nil {
+		return *template.McacEnabled
+	}
+	return true
+}
+
+func setMcacDisabled(dc *cassdcapi.CassandraDatacenter, template *DatacenterConfig) {
+	if !isMcacEnabled(template) {
```

Comment:
Nit: I'd rather pull this if check out of this function since it is surprising to test whether X is enabled or not inside a function whose name is `setXDisabled`.